playFeature/src/main/java/com/panhuk/playfeature/PlayViewModel.kt

```diff
@@ -64,20 +94,42 @@ class PlayViewModel @Inject constructor(
     questions.peek()?.run {
       title = questionTitle
       questionAnswers = allAnswers
+      currentQuestionNumber++
     }
   }
 
-  fun checkAnswer(answer: String): Boolean {
+  fun checkAnswer(answer: String = ""): Boolean {
     val correctAnswer = questions.peek()?.correctAnswer
-    questions.poll()
+    deleteThisQuestion()
+    isLastQuestion()
     loadQuestion()
+    reInitTimer()
 
-    return if (correctAnswer == answer) {
-      totalScore++
-      true
-    } else {
-      false
+    return when (correctAnswer) {
+      answer -> {
+        totalScore++
+        true
+      }
+      else -> false
     }
```

Comment:
I bet you need to improve this method. 
Check your logic and do not perform unnecessary work after certain conditions are met.